Sources/NIOHTTP1Server/main.swift

```diff
@@ -33,6 +33,18 @@ extension String {
     }
 }
 
+private func httpResponseHead(request: HTTPRequestHead, status: HTTPResponseStatus, headers: HTTPHeaders = HTTPHeaders()) -> HTTPResponseHead {
+    var head = HTTPResponseHead(version: request.version, status: status, headers: headers)
+    if request.isKeepAlive && request.version.major == 1 && request.version.minor == 0 {
+        /* to work around `ab -k` behaving weirdly: it'll send an HTTP/1.0 request with keep-alive set
+           but stops doing anything at all if the server doesn't also set Connection: keep-alive */
+        if head.headers["connection"].count == 0 {
```

Comment:
maybe use `isEmpty` ?

Reply:
Also this check may not be good enough as in theory it could still include a "connection" header with another value then `keep-alive`, which means you also need to check the value in the first place

Reply:
@normanmaurer that is for the response so if the user has already set a `connection` header I don't want to override it, does that make sense?